modules/402-ingress-nginx/templates/kruise/webhook.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: kruise-mutating-webhook-configuration
+  {{- include "helm_lib_module_labels" (list . ) | nindent 2 }}
+  annotations:
+    template: ""
+webhooks:
+  - clientConfig:
+      caBundle: Cg==
+      service:
+        name: kruise-webhook-service
+        namespace: d8-ingress-nginx
+        path: /mutate-apps-kruise-io-v1alpha1-daemonset
+    failurePolicy: Fail
+    admissionReviewVersions:
+      - v1
+    sideEffects: None
+    timeoutSeconds: 30
+    name: mdaemonset.kb.io
+    rules:
+      - apiGroups:
+          - apps.kruise.io
+        apiVersions:
+          - v1alpha1
+        operations:
+          - CREATE
+          - UPDATE
+        resources:
+          - daemonsets
+  - clientConfig:
+      caBundle: Cg==
+      service:
+        name: kruise-webhook-service
+        namespace: d8-ingress-nginx
+        path: /mutate-pod
+    failurePolicy: Ignore
+    admissionReviewVersions:
+      - v1
+    sideEffects: None
+    timeoutSeconds: 30
+    name: mpod.kb.io
+    namespaceSelector:
+      matchExpressions:
+        - key: control-plane
+          operator: DoesNotExist
+    rules:
+      - apiGroups:
+          - ""
+        apiVersions:
+          - v1
+        operations:
+          - CREATE
+        resources:
+          - pods
+---
+# Source: kruise/templates/webhookconfiguration.yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: kruise-validating-webhook-configuration
+  {{- include "helm_lib_module_labels" (list . ) | nindent 2 }}
+  annotations:
+    template: ""
+webhooks:
+  - clientConfig:
+      caBundle: Cg==
```

Comment:
Why is this hard coded? Let's generate certificates for webhook as we usually do for other webhooks.

Reply:
We are not able too. These webhooks and secret are managed by kruise controller and it's not configurable. So I left default chart value here

Reply:
Ok, it seems there is no way to play around. Could you add a comment that kruise does not allow configuring of custom certificates and controls them internally (I wonder how it works with helm, shouldn't it redeploy webhook configs on module changes).